### DISCUSSION TITLE
How to use npm (Next.js)

### DISCUSSION THREAD

**Discussion by Helder-SR** [hakimel/reveal.js#3333] [Q&A] [answered]
I'm using the npm for my Next.js App, but I don't understand how to do to display the slides. When I try to display, anything appears.
 Here's my code:
My component:
```

//Slide.js 
import Reveal from 'reveal.js';
import Markdown from 'reveal.js/plugin/markdown/markdown.esm.js';

import '/node_modules/reveal.js/dist/reveal.css';
import '/node_modules/reveal.js/dist/theme/black.css';

Reveal.initialize({
    controls:true,
    width:1000,
    height:1000,
    margin: 0.1,
    display:true,

    plugins: [ Markdown ]

});

export default function Slide(){

    return(
        <>
            <div class="reveal">
			<div class="slides">
				<section>Slide 1</section>
                                  <section data-state="make-it-pop">
                                      <section>Vertical Slide 1</section>
                                      <section>Vertical Slide 2</section>
                                  </section>
                                  <section data-markdown>
                                      <textarea data-template>
                                        ## Slide 1
                                        A paragraph with some text and a [link](http://hakim.se).
                                        ---
                                        ## Slide 2
                                        ---
                                        ## Slide 3
                                      </textarea>
                                  </section>
			</div>
		</div>
        </>    
    )
}
```
And here the page where it should be displayed:
```
//index.js
import dynamic from 'next/dynamic';

const Slide = dynamic(() => import('./slide'), { ssr: false, })

export default function Home() {
    return (
        <div>
            <Slide></Slide>
        </div>
    )
}
```

**Comment by ajwtech** [comment, marked as answer]
Do you have a repo link? I tried to recreate this but I get other errors with next and reveal.

**Reply by Helder-SR** [reply]
Here it is: https://github.com/Helder-SR/nextjs-reveal.git
I found this piece of code that can help but it makes me errors, I don't know if it could help you
```
// Slide.js
export default function Slide(){
    useEffect(() => {
        Reveal.initialize({
            controls:true,
            width:1000,
            height:1000,
            margin: 0.1,
            display:true,
            plugins: [ Markdownu ]
        });
    }, []);

    return(
    …
```

**Reply by ajwtech** [reply]
More details are helpful. what is the new error? 
looks like you have some subfolders in your pages, what address are you trying to hit?
you have a typo on the code you posted.` plugins: [ Markdownu ]` should be `plugins: [Markdown]`
the deck needs to be initialized inside a function or it isnt kept in scope
You also need in instantiate the deck by creating a new object
The rest of the issues are going to be css related I believe. use the developertools and look at the 3d view. you will notice the the deck is physically off the page. 

`<div class="aria-status" aria-live="polite" aria-atomic="true" style="position: absolute; height: 1px; width: 1px; overflow: hidden; clip: rect(1px, 1px, 1px, 1px);">Slide 1 </div>`  is causing it. Not sure where that is coming from but if you remove that style you can see the deck. 

![image](https://user-images.githubusercontent.com/6678192/206435066-37e7d0ed-edaa-4c9c-8504-97ef174cb93f.png)

**Reply by Helder-SR** [reply]
I'm trying to hit the "localhost:3000/slides/slide1". I changed my code, and pushed it (https://github.com/Helder-SR/nextjs-reveal.git), but it's the same error: `
NoModificationAllowedError: Failed to set the 'outerHTML' property on 'Element': This element has no parent node.`
I don't understand where is the problem. And if we remove the parts with `Markdown`:
```
<div class="reveal">
	<div class="slides">
		<section>Slide 1</section>
                <section>
                    <section>Vertical Slide 1</section>
                    <section>Vertical Slide 2</section>
                </section>             
	</div>
</div>
```
I get this error: `NotFoundError: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.`
I don't know how did you do to display the slide but if you could make a repo it would be very helpful.

**Reply by Helder-SR** [reply]
I haven't touch anything and now it works, I really don't understand why, but thanks for your help!
